internal/cephfs/controllerserver.go

```diff
@@ -800,6 +800,16 @@ func (cs *ControllerServer) CreateSnapshot(
 			}
 		}
 
+		// Update snapshot-name/snapshot-namespace/snapshotcontent-name details on
+		// subvolume snapshot as metadata in case snapshot already exist
+		metadata := k8s.GetSnapshotMetadata(req.GetParameters())
+		if len(metadata) != 0 {
+			err = snapClient.SetAllSnapshotMetadata(metadata)
+			if err != nil {
+				return nil, status.Error(codes.Internal, err.Error())
+			}
+		}
```

Comment:
skip this one if someone is using an older version of ceph which does not support metadata operation?

Reply:
Done now, PTAL.